lib/Http/XMLResponse.php

```diff
@@ -136,8 +136,16 @@ public function arrayToXml(array $data, ?string $rootTag = null): string
 		// Build XML structure
 		$this->buildXmlElement($dom, $root, $data);
 		
-		// Convert DOM to string
-		return $dom->saveXML() ?: '';
+		// Get XML output
+		$xmlOutput = $dom->saveXML() ?: '';
+		
+		// Directly replace decimal CR entities with hexadecimal
+		$xmlOutput = str_replace('&#13;', '&#xD;', $xmlOutput);
```

Comment:
Are these html encoded characters? Should these not be done with mapping? Feels risky to replace all perhaps

Reply:
Actually, these are equivalent characters, &#13 and &#xD both resolve to the same character, only &#13 is the decimal value, and &#xD is the hexadecimal value.
The best way would be if we were able to choose the type of html encoding (decimal or hexadecimal), so for now I'd say this is fine.